### DISCUSSION TITLE
[BUG] setproctitle and timelib compilation requirements on the Windows platform

### DISCUSSION THREAD

**Discussion by arizvisa** [saltstack/salt#67299] [Feature Requests]
**Description**
On the Windows platform, both of these packages are required if installing the `salt==3001` package from pypi (https://pypi.org/project/salt/3001). The issue is that installing either of these packages requires a compiler on the target in order to build them.

However, according to the `HAS_TIMELIB` and `HAS_SETPROCTITLE` variables used by `salt.modules.dockermod` + `salt.utils.dateutils`, and `salt.utils.process` (respectively), these modules are actually optional.

Specifically wrt`setproctitle`, the `setproctitle` module does not have the same semantics on Windows at all. Namely it creates an event handle with a name, and is then discarded (it's assigned at the function-static scope and not returned in any form). Nothing in salt seems to expose this (nor do many consumers of the library explain how to locate the commandline parameters for the process). This means that to effectively use this capability in the way intended, you'll need to understand a little bit of how handles work in Windows and how to enumerate them through the object namespace. This could be documented for normies, but it isn't.

**Setup**
Install the `salt` module with pip. Constrain it to version `3001`. Make sure there's no compiler so that way it can fail.

**Steps to Reproduce the behavior**
`pip install salt==3001`.

**Expected behavior**
As salt-minions are pretty much pure-python, a compiler in the Windows environment shouldn't be necessary to install salt...and again, they were both actually considered optional according to the code.

**Screenshots**
None.

**Versions Report**
3001 as per https://pypi.org/project/salt/3001/

**Comment by arizvisa** [comment]
I can get logs if you guys need, lmk if that's the case.

I think this is actually in `requirements/static/windows.in`.

**Comment by krionbsd** [comment]
Shane, Chad, could you please help with it?
